pkg/detector/detector.go

```diff
@@ -720,26 +720,9 @@ func (d *ResourceDetector) GetUnstructuredObject(objectKey keys.ClusterWideKey)
 	return unstructuredObj, nil
 }
 
-func (d *ResourceDetector) getPropagationPolicyID(policy *policyv1alpha1.PropagationPolicy) (string, error) {
-	id := util.GetLabelValue(policy.GetLabels(), policyv1alpha1.PropagationPolicyPermanentIDLabel)
-	if id == "" {
-		id = uuid.New().String()
-		policy.Labels = util.DedupeAndMergeLabels(policy.Labels, map[string]string{policyv1alpha1.PropagationPolicyPermanentIDLabel: id})
-		if err := d.Client.Update(context.TODO(), policy); err != nil {
-			return id, err
-		}
-	}
-
-	return id, nil
-}
-
 // ClaimPolicyForObject set policy identifier which the object associated with.
 func (d *ResourceDetector) ClaimPolicyForObject(object *unstructured.Unstructured, policy *policyv1alpha1.PropagationPolicy) (string, error) {
-	policyID, err := d.getPropagationPolicyID(policy)
-	if err != nil {
-		klog.Errorf("Get PropagationPolicy(%s/%s) ID error:%v", policy.Namespace, policy.Name, err)
-		return "", err
-	}
+	policyID := policy.Labels[policyv1alpha1.PropagationPolicyPermanentIDLabel]
```

Comment:
Have you ever thought about what's gonna happen if there is no `permanent` ID?
I know it is not a normal case, but just guess, is there something bad that would happen? And is it easy to identify that the issue is caused by the lack of permanent ID?

Reply:
good question. 

> is there something bad that would happen?

this is very likely to cause the resource template to be unable to be properly unbound from the policy.

> And is it easy to identify that the issue is caused by the lack of permanent ID?

it's very hard.

since the policy is a user-oriented resource, can we introduce a protection mechanism for the permanent ID? For example, add verification to the webhook to protect this label from being modified or deleted.

Reply:
+1
on the webhook validation.